tests/FormlyMapper/FormlyField/BirthdayFieldTest.php

```diff
@@ -0,0 +1,287 @@
+{
+    /**
+     * @var DateField
+     */
+    protected $formlyField;
+
+    public function basicDataProvider(): array
+    {
+        $config = [
+            'name' => 'borndate',
+            'type' => 'birthday',
+            'options' => [
+                'required' => true,
+                'label' => 'Borndate',
+            ],
+        ];
+
+        $expected = [
+            'key' => 'borndate',
+            'type' => 'input',
+            'templateOptions' => [
+                'required' => true,
+                'label' => 'Borndate',
+                'type' => 'birthday',
+            ],
+        ];
+
+        return [
+            [$config, $expected],
+        ];
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testIsAddingBirthdayFields(array $fieldConfiguration, array $expected): void
+    {
+        $fieldConfiguration['validation'] = [
+            'Symfony\Component\Validator\Constraints\Regex' => [
+                'pattern' => '^[0-9]{4}\-[0-9]{2}\-[0-9]{2}$',
+                'message' => 'The birthday field must follow the pattern "yyyy-MM-dd"',
+            ],
+        ];
+
+        $expected['templateOptions']['pattern'] = '^[0-9]{4}\-[0-9]{2}\-[0-9]{2}$';
+        $expected['validation'] = [
+            'messages' => [
+                'regex' => 'The birthday field must follow the pattern "yyyy-MM-dd"',
+            ],
+        ];
+
+        $this->formlyField->setFieldConfiguration($fieldConfiguration);
+        $actual = $this->formlyField->getFormlyFieldConfiguration();
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testIsAddingSpecificYears(array $fieldConfiguration, array $expected): void
+    {
+        $years = [2000, 2001, 2002, 2003, 2004, 2005];
+
+        $fieldConfiguration['options']['years'] = $years;
+
+        $expected['templateOptions']['years'] = $years;
+
+        $this->formlyField->setFieldConfiguration($fieldConfiguration);
+        $actual = $this->formlyField->getFormlyFieldConfiguration();
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testWithNotNumberSpecificYears(array $fieldConfiguration, array $expected): void
+    {
+        $goodYears = [2000, 2001, 2002, 2003, 2004, 2005];
+        $badYears = ['2000', 2001, '2002', 2003, 2004, 2005];
+
+        $fieldConfiguration['options']['years'] = $badYears;
+
+        $expected['templateOptions']['years'] = $goodYears;
+
+        $this->formlyField->setFieldConfiguration($fieldConfiguration);
+        $actual = $this->formlyField->getFormlyFieldConfiguration();
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testIsAddingYearsRangeInOrderByDefault(array $fieldConfiguration, array $expected): void
```

Comment:
Nothing is happening by default though. What I think you meant to say is `InDefaultOrder` not `InOrderByDefault`.

Reply:
Oh yeah, you got it right. Sorry I'm trying to improve my English every day :)